src/uuid/builder/uuidbuilder.rs

```diff
@@ -165,15 +96,15 @@ impl UUIDv8Builder {
                 }
             } else {
                 self.msb
-                    .store((instant << 16) | (8u64 << 12), Ordering::SeqCst);
+                    .store(instant << 16, Ordering::SeqCst);
```

Comment:
I'd like a little help sanity checking here. I'm trying to understand what's typical since I've not explored the internals of middleware software before.

It seems like with 48 bits, we'd be good till over 8,000 years past the Unix epoch if we count in milliseconds.

I'm aware that it's pretty ambitious to say that this protocol will exist in over 8,000 years, so would it be more typical to say, leave a comment, something to the effect of:
```rust
// 48 bits for ms since Unix epoch in roughly year 10889,  so we assume that 
// it is before then and do not check whether we have used past 48 bits
```
or would we want to be thorough and include a check for this?
```rust
// ...
// -- this is new start --
const MAX_TIMESTAMP_BITS: u8 = 48;
const MAX_TIMESTAMP_MASK: u64 = 0xffff << MAX_TIMESTAMP_BITS;
// -- this is new end --
// ...
    pub fn build_with_instant(&self, instant: u64) -> uproto_Uuid {
        let new_msb = {
            let current_msb = self.msb.load(Ordering::SeqCst);

            if instant == (current_msb >> 16) {
                if (current_msb & MAX_COUNT) < MAX_COUNT {
                    self.msb.fetch_add(1, Ordering::SeqCst);
                }
            } else {
                // -- this is new start --
                if instant & MAX_TIMESTAMP_MASK != 0 {
                    panic!("Overflowed 48 bits of unit_ts_ms")
                }
                // -- this is new end --
                self.msb
                    .store(instant << 16, Ordering::SeqCst);
            }

            self.msb.load(Ordering::SeqCst)
        };

        let mut bytes = [0u8; 16];
        bytes[..8].copy_from_slice(&new_msb.to_be_bytes());
        bytes[8..].copy_from_slice(&self.lsb);
        Builder::from_custom_bytes(bytes).into_uuid().into()
    }
}
```
and the accompanying test:
```
    #[test]
    #[should_panic(expected = "Overflowed 48 bits of unit_ts_ms")]
    fn test_uuid_past_max_unit_ts_ms () {
        // maximum value that can be stored in a 48-bit timestamp (in milliseconds)
        let max_48_bit_unix_ts_ms = (1u128 << 48) - 1;

        // add 1 millisecond to the maximum duration, to overflow
        let overflowed_48_bit_unix_ts_ms = (max_48_bit_unix_ts_ms) as u64 + 1;

        let builder = UUIDv8Builder::new();
        let _uprotocol_uuid_past_max_unix_ts_ms = builder.build_with_instant(overflowed_48_bit_unix_ts_ms);
    }
```

I suppose since we expose `UUIDv8Builder::build_with_instant()` to the user / implementor they could call it with an arbitrary `u64` that'd overflow the 48 bits we have. (Maybe a side-topic, but we'd avoid at least that issue if it were not public, but that may have other ramifications)

Given that creating UUIDs is pretty integral to uProtocol and the 8,000 years away that is, is it worth it to pay the (admittedly very small) overhead of a bitwise and for each and every UUID built?

A short not-entirely-representative example of with and without the panic on Compiler Explorer does show the extra assembly for the panic, but the times are identical when looking at the output: [Compiler Explorer example](https://godbolt.org/z/E743eET4r)

Is panicking the right thing to do in that case to indicate failure?

Sorry for writing a short story here! Hope to hear your thoughts.

Reply:
I do share your concern regarding the potential overflow in general. 

> Is panicking the right thing to do in that case to indicate failure?

It could be, if we properly document that it does so. Panicking would have the advantage of letting client code decide if and how to make sure that the passed in value is acceptable. Some might want to check explicitly, others might be able to rely on the fact that it never does due to the way the value has been created.

On the other hand, we could also simply play it safe, change the return type to `Result<uproto_Uuid, ()> and return an Err if the passed in instant is longer than 48 bits.

FMPOV the former alternative is preferable because it preserves the simplicity of using the function (no result checking necessary). I also believe that in almost all cases, the `build()` function will be used instead, i.e. using the current time instead of a passed in instant.

WDYT?

Reply:
For my 2c, I like keeping the return type clean for the 99.9999% of cases where the timestamp would fit in 48 bits and adding some documentation that it can panic.

Perhaps the unit test I wrote in the comment could become a doc comment to illustrate when this would occur?